### PR TITLE
docs: add support for model providers with usage examples

### PR DESCRIPTION
**Pull request by Haleshot**
## 📝 Summary

Update the `chat.md` file to include examples for GROQ and Cerebras model providers (while also linking the [LiteLLM page](https://litellm.vercel.app/docs/providers) to showcase various other model providers).
Address comment - https://github.com/marimo-team/marimo/issues/2773#issuecomment-2455326251

## 🔍 Description of Changes

- Added examples for GROQ and Cerebras model providers.
- Included a note about the availability of free API keys for GROQ and Cerebras.
- Provided guidance on finding the correct endpoint URLs for model providers (refer to docs; usually found in the curl commands).

## 📋 Checklist

- [x] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [ ] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://marimo.io/discord?ref=pr), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [ ] I have added tests for the changes made.
- [x] I have run the code and verified that it works as expected.

## 📜 Reviewers
@akshayka OR @mscolnick

### CODE REVIEW

File: docs/api/inputs/chat.md [hunk 1]
```diff
@@ -37,6 +37,88 @@ AI models from popular providers or custom functions.
 
 ## Basic Usage
 
+## Supported Model Providers
```
Comment (mscolnick) [inline docs/api/inputs/chat.md:40]: can this go at the bottom?

File: docs/api/inputs/chat.md [hunk 2]
```diff
@@ -257,4 +339,3 @@ configuration with a dictionary conforming to the config.
 
 ```{eval-rst}
 .. autoclass:: marimo.ai.ChatAttachment
-```
```
Comment (Haleshot) [inline docs/api/inputs/chat.md:260]: This deletion doesn't seem to be breaking `ChatAttachment` class from rendering in the docs deployed; hope this is alright?

File: docs/api/inputs/chat.md (hunk 1 above)
Comment (Haleshot) [reply docs/api/inputs/chat.md:40]: > can this go at the bottom?

Sure, will reposition this.

File: docs/api/inputs/chat.md [hunk 3]
```diff
@@ -37,6 +37,88 @@ AI models from popular providers or custom functions.
...
+   ),
+)
+chatbot
+```
+
+:::
+::::
+
+```{note}
+We have added examples for GROQ and Cerebras. These providers offer free API keys and are great for trying out Llama models (from Meta). Cerebras, as of today, offers the fastest inferencing of the Llama3.1 70B model and GROQ is also competitive in this space.
```
Comment (mscolnick) [inline docs/api/inputs/chat.md:78]: I think we can drop `Cerebras, as of today,...competitive in this space.`

File: docs/api/inputs/chat.md [hunk 4]
```diff
@@ -37,6 +37,88 @@ AI models from popular providers or custom functions.
...
+   mo.ai.llm.openai(
+       model="llama3.1-8b",
+       api_key=key,
+       base_url="https://api.cerebras.ai/v1/",
+   ),
+)
+chatbot
+```
+
+We recommend using free providers like [GROQ](https://groq.com/) and [Cerebras](https://cerebras.ai/), as you can sign up on their platforms and integrate with various AI integrations in marimo easily. For more information, refer to the [AI completion documentation in marimo](https://docs.marimo.io/guides/editor_features/ai_completion.html#ai-completion).
```
Comment (mscolnick) [inline docs/api/inputs/chat.md:98]: this could be `/guides/editor_features/ai_completion#ai-completion`

Comment (mscolnick) [inline docs/api/inputs/chat.md:98]: i think you pretty much mention this in the `note` above. i would either consolidate or remove this one

File: docs/api/inputs/chat.md [hunk 5]
```diff
@@ -258,3 +258,83 @@ configuration with a dictionary conforming to the config.
...
+       api_key=key,
+       base_url="https://api.cerebras.ai/v1/",
+   ),
+)
+chatbot
+```
+
+If you want more examples for model providers and their endpoints listed, please submit a [feature request](https://github.com/marimo-team/marimo/issues/new?template=documentation.yaml).
+
+### Example Usage
```
Comment (mscolnick) [inline docs/api/inputs/chat.md:322]: i think we can delete these, we already have the same example above

File: docs/api/inputs/chat.md [hunk 6]
```diff
@@ -258,3 +258,83 @@ configuration with a dictionary conforming to the config.
...
+       model="llama-3.1-70b-versatile",
+       api_key="gsk_...", # insert your key here,
+       base_url="https://api.groq.com/openai/v1/",
+   ),
+)
+chatbot
+```
+
+:::
+::::
```
Comment (mscolnick) [inline docs/api/inputs/chat.md:297]: can you add an entry for xai? then we can close out that ticket #2773 
```
model="grok-beta",
base_url="https://api.x.ai/v1",
```

File: docs/api/inputs/chat.md (hunk 5 above)
Comment (Haleshot) [reply docs/api/inputs/chat.md:322]: Ah alright; my reasoning was, the first link above (is for a feature request) to include model providers which can't be run through overriding the `base_url` -> not openAI compatible; the hyperlink below is for a doc change (.doc.yaml) if people want more model providers to be listed in this chat.md doc webpage (for reference).

File: docs/api/inputs/chat.md (hunk 6 above)
Comment (mscolnick) [reply docs/api/inputs/chat.md:297]: i pulled this from their docs, i think its ok to not test and assume their docs are accurate in that case

**Review by mscolnick — APPROVED** [review]
looks great, thank you!

**Pull request merged**